### PR TITLE
Joke Icons

### PR DESCRIPTION
**Pull request by Vsumskiy**

### CODE REVIEW

File: src/app/directives/input-upload.directive.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    public element: ElementRef,
+    private jokesMediaService: JokesMediaService
+  ) {}
+
+  ngOnInit(): void {
+    fromEvent(this.element.nativeElement, 'change')
+      .pipe(map((i: InputEvent) => (i.target as HTMLInputElement).files))
+      .subscribe(icons => {
+        const imagesExtension = Object.values(icons).reduce(
+          (acc, curr: File) => {
```
Comment (iyarymovych-reenbit) [inline src/app/directives/input-upload.directive.ts:21]: `acc`, `curr` - rename them to more understandable and semantic names.

File: src/app/directives/input-upload.directive.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+            acc.push(this.jokesMediaService.getImageData(extension));
+            return acc;
+          },
+          []
+        );
+        zip(...imagesExtension).subscribe(
+          (data: ImageInterface[]) => {
+            this.jokesMediaService.setImageData(data);
+          },
+          e => {}
```
Comment (iyarymovych-reenbit) [inline src/app/directives/input-upload.directive.ts:32]: Unneeded error handler.

File: src/app/pages/category-customization/category-form/category-form.component.html [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    [formGroup]="form"
+    class="form-modify"
+    (submit)="submit()">
+
+    <div class="form-group"
+        *ngIf="addCategoryMode">
+        <mat-form-field appearance="legacy">
+            <mat-label>Add new category</mat-label>
+            <input matInput
+                class="h"
```
Comment (iyarymovych-reenbit) [inline src/app/pages/category-customization/category-form/category-form.component.html:11]: What does `h` mean?

File: src/app/validators/checkPasswords.validator.ts [hunk 4]
```diff
@@ -0,0 +1,13 @@
+import { FormGroup, ValidationErrors } from '@angular/forms';
+
+export class PasswordsValidator {
```
Comment (iyarymovych-reenbit) [inline src/app/validators/checkPasswords.validator.ts:3]: Why do you need class for custom validator?

It should be just a function.

File: src/app/validators/checkPasswords.validator.ts [hunk 5]
```diff
@@ -0,0 +1,13 @@
+import { FormGroup, ValidationErrors } from '@angular/forms';
```
Comment (iyarymovych-reenbit) [inline src/app/validators/checkPasswords.validator.ts:1]: Create only one service that holds all validators that possibly can be reused.

File: src/app/directives/input-upload.directive.ts (hunk 1 above)
Comment (Vsumskiy) [reply src/app/directives/input-upload.directive.ts:21]: done!

File: src/app/directives/input-upload.directive.ts (hunk 2 above)
Comment (Vsumskiy) [reply src/app/directives/input-upload.directive.ts:32]: done

File: src/app/validators/checkPasswords.validator.ts (hunk 4 above)
Comment (Vsumskiy) [reply src/app/validators/checkPasswords.validator.ts:3]: Done

**Pull request merged**